pennylane/ops/op_math/sprod.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the SProd class which contains logic for
+computing the scalar product of operations.
+"""
+import pennylane as qml
+from .symbolicop import SymbolicOp
+
+
+def s_prod(scalar, operator, do_queue=True, id=None):
+    """Associate an operator with a scalar to represent scalar multiplication."""
+    return SProd(scalar, operator, do_queue=do_queue, id=id)
+
+
+def _sprod(mat, scalar, dtype=None, cast_like=None):
+    """Multiply matrix with scalar"""
+    res = qml.math.multiply(scalar, mat)
+
+    if dtype is not None:  # additional casting logic
+        res = qml.math.cast(res, dtype)
+    if cast_like is not None:
+        res = qml.math.cast_like(res, cast_like)
+
+    return res
+
+
+class SProd(SymbolicOp):
+    """Arithmetic operator subclass representing the scalar product of an
+    operator with the given scalar."""
+
+    def __init__(self, scalar, base, do_queue=True, id=None):
+        self.scalar = scalar
+        super().__init__(base=base, do_queue=do_queue, id=id)
+        self._name = "SProd"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return f"{self.scalar}*({self.base})"
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        """The label produced for the SProd op."""
+        scalar_val = f"{self.scalar}"
+        if decimals is not None:
+            scalar_val = format(qml.math.toarray(self.scalar), f".{decimals}f")
+
+        return base_label or f"{scalar_val}*{self.base.label(decimals=decimals, cache=cache)}"
+
+    @property
+    def data(self):
+        """The trainable parameters"""
+        return [[self.scalar], self.base.data]  # Not sure if this is the best way to deal with this
```

Comment:
Would love to get your opinion on this? Is there a better way to pass the scalar value to the data property?

Reply:
I don't think there's a definitive best way to do this.  We can probably just choose one way and move forward with it till we get more information later.  

This format is more similar to how we did it in `Sum`, but it will lead to a lot more nesting. We could also just go with 
`[self.scalar] + self.base.data`.  That would keep the data list flatter.